
local input = {}

type KeyboardKey = number

function input.haspressed(key: KeyboardKey): boolean
    error("Not implemented")
end

function input.isdown(key: KeyboardKey): boolean
    error("Not implemented")
end

input.keys = {
    ["null"] = 0,
    ["apostrophe"] = 0,
    ["comma"] = 0,
    ["minus"] = 0,
    ["period"] = 0,
    ["slash"] = 0,
    ["zero"] = 0,
    ["one"] = 0,
    ["two"] = 0,
    ["three"] = 0,
    ["four"] = 0,
    ["five"] = 0,
    ["six"] = 0,
    ["seven"] = 0,
    ["eight"] = 0,
    ["nine"] = 0,
    ["semicolon"] = 0,
    ["equal"] = 0,
    ["a"] = 0,
    ["b"] = 0,
    ["c"] = 0,
    ["d"] = 0,
    ["e"] = 0,
    ["f"] = 0,
    ["g"] = 0,
    ["h"] = 0,
    ["i"] = 0,
    ["j"] = 0,
    ["k"] = 0,
    ["l"] = 0,
    ["m"] = 0,
    ["n"] = 0,
    ["o"] = 0,
    ["p"] = 0,
    ["q"] = 0,
    ["r"] = 0,
    ["s"] = 0,
    ["t"] = 0,
    ["u"] = 0,
    ["v"] = 0,
    ["w"] = 0,
    ["x"] = 0,
    ["y"] = 0,
    ["z"] = 0,
    ["left_bracket"] = 0,
    ["backslash"] = 0,
    ["right_bracket"] = 0,
    ["grave"] = 0,
    ["space"] = 0,
    ["escape"] = 0,
    ["enter"] = 0,
    ["tab"] = 0,
    ["backspace"] = 0,
    ["insert"] = 0,
    ["delete"] = 0,
    ["right"] = 0,
    ["left"] = 0,
    ["down"] = 0,
    ["up"] = 0,
    ["page_up"] = 0,
    ["page_down"] = 0,
    ["home"] = 0,
    ["end"] = 0,
    ["caps_lock"] = 0,
    ["scroll_lock"] = 0,
    ["num_lock"] = 0,
    ["print_screen"] = 0,
    ["pause"] = 0,
    ["f1"] = 0,
    ["f2"] = 0,
    ["f3"] = 0,
    ["f4"] = 0,
    ["f5"] = 0,
    ["f6"] = 0,
    ["f7"] = 0,
    ["f8"] = 0,
    ["f9"] = 0,
    ["f10"] = 0,
    ["f11"] = 0,
    ["f12"] = 0,
    ["left_shift"] = 0,
    ["left_control"] = 0,
    ["left_alt"] = 0,
    ["left_super"] = 0,
    ["right_shift"] = 0,
    ["right_control"] = 0,
    ["right_alt"] = 0,
    ["right_super"] = 0,
    ["kb_menu"] = 0,
    ["kp_0"] = 0,
    ["kp_1"] = 0,
    ["kp_2"] = 0,
    ["kp_3"] = 0,
    ["kp_4"] = 0,
    ["kp_5"] = 0,
    ["kp_6"] = 0,
    ["kp_7"] = 0,
    ["kp_8"] = 0,
    ["kp_9"] = 0,
    ["kp_decimal"] = 0,
    ["kp_divide"] = 0,
    ["kp_multiply"] = 0,
    ["kp_subtract"] = 0,
    ["kp_add"] = 0,
    ["kp_enter"] = 0,
    ["kp_equal"] = 0,
    ["back"] = 0,
    ["menu"] = 0,
    ["volume_up"] = 0,
    ["volume_down"] = 0,
} 

return input
